src/mcts/search.cc

```diff
@@ -187,45 +187,77 @@ void Search::SendMovesStats() const {
   const float parent_q =
       -root_node_->GetQ(0) -
       kFpuReduction * std::sqrt(root_node_->GetVisitedPolicy());
+  const float U_coeff =
+      kCpuct * std::sqrt(std::max(root_node_->GetChildrenVisits(), 1u));
+
   for (Node* child : root_node_->Children()) {
     nodes.emplace_back(child);
   }
   std::sort(nodes.begin(), nodes.end(),
-            [](const Node* a, const Node* b) { return a->GetN() < b->GetN(); });
+           [&parent_q, &U_coeff](const Node* a, const Node* b) {
+             return
+                 std::forward_as_tuple(a->GetN(), a->GetQ(parent_q) + U_coeff * a->GetU())
+               < std::forward_as_tuple(b->GetN(), b->GetQ(parent_q) + U_coeff * b->GetU());
+           });
 
   const bool is_black_to_move = played_history_.IsBlackToMove();
   ThinkingInfo info;
   for (const Node* node : nodes) {
     std::ostringstream oss;
     oss << std::fixed;
+
     oss << std::left << std::setw(5)
         << node->GetMove(is_black_to_move).as_string();
+
     oss << " (" << std::setw(4) << node->GetMove().as_nn_index() << ")";
-    oss << " N: ";
-    oss << std::right << std::setw(7) << node->GetN() << " (+" << std::setw(2)
-        << node->GetNInFlight() << ") ";
-    oss << "(V: " << std::setw(6) << std::setprecision(2) << node->GetV() * 100
-        << "%) ";
+
+    oss << " N: " << std::right << std::setw(7) << node->GetN() << " (+"
+        << std::setw(2) << node->GetNInFlight() << ") ";
+
+    oss << "(V: ";
+    // Handrolled optional only has a few overloads available, simple usage only
```

Comment:
You know, I tried it in a slightly different way the first time, but it didn't work quite right because of the limited functionality of the handrolled optional, so I changed it to this which works.

But looking again, why didn't I just do this?
```
optional<float> v;
if (node->IsTerminal()) {
  v = node->GetTerminalNodeValue();
} else { 
  v = GetCachedFirstPlyValue(node);
}
```

?

I think the answer is that I was trying to combine that ^ set of conditionals with the ones below about the actual string, but now I'm thinking this is best.

Reply:
Looks good.